src/MonitoringFunctions/Functions/WorkItemCreator.cs

```diff
@@ -0,0 +1,187 @@
+using MonitoringFunctions.Models;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MonitoringFunctions.Functions
+{
+    internal static class WorkItemCreator
+    {
+        private const string _projectName = "devdiv";
+        private static readonly string? _devdivAdoPAT = Environment.GetEnvironmentVariable("devdiv-ado-pat");
+        private static readonly Uri _devdivCollectionUri = new Uri("https://devdiv.visualstudio.com/DefaultCollection/");
+
+        [FunctionName("WorkItemCreator")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)] HttpRequest req,
+            ILogger log)
+        {
+            List<WorkItem> createdWorkItems = new List<WorkItem>();
+
+            using StreamReader requestStream = new StreamReader(req.Body);
+            string requestBody = await requestStream.ReadToEndAsync().ConfigureAwait(false);
+
+            if(string.IsNullOrWhiteSpace(requestBody))
+            {
+                return new BadRequestObjectResult("Request body shouldn't be empty");
+            }
+
+            AlertNotificationData data;
+            try
+            {
+                data = JsonConvert.DeserializeObject<AlertNotificationData>(requestBody);
+            }
+            catch(JsonException e)
+            {
+                log.LogError(e, "Deserialization of the request body has failed.");
+                return new BadRequestObjectResult("Deserialization of the request body has failed.");
+            }
+
+            string alertMessage = data.Message ?? "Alert triggered";
+            int alertingMonitorCount = data.MatchingAlerts?.Length ?? 0;
+
+            if(data.State != AlertState.Alerting)
+            {
+                log.LogError("Alert state is not \"Alerting\". This notification shouldn't have been sent to this function.");
+                return new BadRequestObjectResult("Alert state is not \"Alerting\". This notification shouldn't have been sent to this function.");
+            }
+
+            for(int i=0; i<alertingMonitorCount; i++)
+            {
+                AlertEvaluation? match = data.MatchingAlerts![i];
+
+                if(match == null || match.Tags == null)
+                {
+                    log.LogError($"Evaluationg match #{i} has invalid data.");
+                    continue;
+                }
+
+                if(!match.Tags.ContainsKey("monitor_name"))
+                {
+                    log.LogError($"Evaluation match #{i} doesn't specify a monitor name which is required.");
+                    continue;
+                }
+
+                string title = $"{match.Tags["monitor_name"]} {alertMessage}";
+                string description = $"Alert details:{Environment.NewLine}{requestBody}";
```

Comment:
should we check if such an ACTIVE incident already exist to avoid creating duplicates?

Reply:
Added the logic to ensure there will be no duplicates.